src/functions.php

```diff
@@ -2,9 +2,72 @@
 
 namespace Amp\Http\Server;
 
+use Amp\ByteStream\InputStream;
 use Amp\Http\Status;
+use Amp\Promise;
+use Amp\Socket\Server as Socket;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use function Amp\call;
 
 function redirectTo(string $uri, int $statusCode = Status::FOUND): Response
 {
     return new Response($statusCode, ['location' => $uri]);
 }
+
+/**
+ * Returns a response with the html content type.
+ *
+ * @param InputStream|string|null $content
+ * @param int $status
+ * @return Response
+ */
+function html($content, int $status = Status::OK): Response
+{
+    return new Response($status, [
+        'Content-Type' => 'text/html; charset=utf-8'
+    ], $content);
+}
+
+/**
+ * Creates an HTTP server.
+ *
+ * This function is inspired in Golang's http module one. It provides sensible defaults for a really quick start
+ * and it is designed to cover 90% of use cases.
+ *
+ * If you need fine-tuning, like listening in two sockets or attach more listeners to the server
+ * you should bootstrap your server manually.
+ *
+ * @param string $address
+ * @param RequestHandler $handler
+ * @param Options|null $options
+ * @param LoggerInterface|null $logger
+ * @return Promise<HttpServer>
+ */
+function listenAndServe(string $address, RequestHandler $handler, ?Options $options, ?LoggerInterface $logger): Promise
+{
+    return call(static function () use ($address, $handler, $options, $logger) {
+        $sockets = [
+            Socket::listen($address)
+        ];
+
+        $logger = $logger ?? new NullLogger();
```

Comment:
I think I'd add logger before options and not make it optional, because it's really helpful to have it enabled. Especially if you're starting and don't know what might be wrong.

Reply:
Well, I thought that at the beginning, but people end up passing the `NullLogger` anyways for a quick start. The optional method does not prevent you to use a logger for debugging if you need to, but it does make the quickstart simple. 😄 

And yeah, I think logger before options is actually better. Found myself passing the logger more than the options.